examples/resource_allocation.ipynb

```diff
@@ -0,0 +1,1495 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Solving Decentralizded Optimal Exchange Problem with BlueFog "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### In this example, you will learn:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. How to solve optimal exchange problem with centralized ADMM;\n",
+    "\n",
+    "\n",
+    "2. How to solve optimal exchange problem with decentralized algorithms such as EXTRA, Exact Diffusion, and Gradient Tracking;\n",
+    "\n",
+    "\n",
+    "3. How to setup topology, determine weights, and run neighbor_allreduce with BlueFog;\n",
+    "\n",
+    "\n",
+    "4. How to use BlueFog functions such as **bf.allreduce(), bf.neighbor_allreduce(), bf.load_topology(), bf.set_topology(), topology_util.GetRecvWeights()**, and **topology_util.ExponentialTwoGraph()**\n",
+    "\n",
+    "\n",
+    "5. How to implement EXTRA, Exact Diffusion, and Gradient Tracking with BlueFog"
+   ]
```

Comment:
Do not have extra newline in middle. Let the markdown give the index automaitically. 
You can check the preview, it is all 1. xx 1.xxx 1.xxx